d2l-dom-visibility.html

```diff
@@ -0,0 +1,50 @@
+<link rel="import" href="d2l-dom.html">
+<script>
+(function() {
+	'use strict';
+
+	var Visibility = {
+
+		isVisible: function(node) {
+
+			if (!node) {
+				return false;
+			}
+
+			if (!node.host) {
```

Comment:
`style` is `undefined` on a shadow-root, hence this check.  In such a case we need to continue walking the tree checking the parents.